dhis-2/dhis-services/dhis-service-dxf2/src/main/java/org/hisp/dhis/dxf2/datavalueset/SpringDataValueSetStore.java

```diff
@@ -374,10 +377,13 @@ private String getAttributeOptionComboClause( User user )
             .collect( Collectors.toList() );
 
         String groupAccessCheck =  groupsIds.size() > 0 ?
-            "or ( " +
-                "check_user_group_ids( co.sharing, '{"+ String.join( ",", groupsIds ) +"}' ) = true " +
-                "and check_user_groups_access( co.sharing, '__r%', '{"+ String.join( ",", groupsIds ) +"}' ) = true ) )"
-            : ") )";
+            " or ( " + JsonbFunctions.HAS_USER_GROUP_IDS + "( co.sharing, '{"+ String.join( ",", groupsIds ) + "}' ) = true " +
```

Comment:
Consider using String.format or similar here as well just to make it a bit more readable.

Reply:
fixed